lib/rspec-puppet/adapters.rb

```diff
@@ -0,0 +1,107 @@
+module RSpec::Puppet
+  module Adapters
+
+    class Base
+      def setup_puppet(example_group)
+        settings_map.each do |puppet_setting, rspec_setting|
+          set_setting(example_group, puppet_setting, rspec_setting)
+        end
+      end
+
+      def set_setting(example_group, puppet_setting, rspec_setting)
+        if example_group.respond_to?(rspec_setting)
+          value = example_group.send(rspec_setting)
+        else
+          value = RSpec.configuration.send(rspec_setting)
+        end
+        begin
+          Puppet[puppet_setting] = value
+        rescue ArgumentError
+          # TODO: this silently swallows errors when applying settings that the current
```

Comment:
Aaaactually... those are also private :/ https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/warnings.rb#L19-L21

From the rspec-support readme:

```
RSpec::Support provides common functionality to RSpec::Core, RSpec::Expectations and RSpec::Mocks. It is considered suitable for internal use only at this time.
```

:/

However, the implementation of warnings is really simple; it's just calling `Kernel.warn` by default (https://github.com/rspec/rspec-support/blob/master/lib/rspec/support.rb#L108-L119). At this point I think it would be less trouble for us to implement whatever warning system we needed.

I'm going to spin out a new issue for this.